src/ptvsd/adapter/messages.py

```diff
@@ -213,6 +233,7 @@ def _configure(self):
 
         # Let the IDE know that it can begin configuring the adapter.
         state.change("configuring")
+        self._set_debugger_properties(self._shared.arguments)
```

Comment:
This should go before the state change, right after the messages are replayed. Just to make sure that we get that message in first before any other comes (the fact that handlers are processed one by one already kinda ensures that, but the state transition makes it clearer).